Adbdev 4793v2

andr-sokolov opened the pull request:
## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

Commit:
Check dynamic_cast result

dynamic_cast result is used without null check. This patch adds an assertion

Commit:
Revert "Check dynamic_cast result"

This reverts commit 9ca4d142b64ff792ec402768bf99c928ee9dd94e.

Commit:
Check dynamic_cast result (#866)

dynamic_cast result is used without null check. This patch adds an assertion

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/65158

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1094330

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1094331

bandetto committed:
Add assertions after dynamic_cast  (#865)

dynamic_cast may return NULL, so assertions were added to catch possible NULL
pointers before dereferencing them.

bandetto committed:
Add an assert for worker (#864)

CWorkerPoolManager::WorkerPoolManager()->Self() always returns CWorker * from
dynamic_cast if the static member is not NULL, because worker pool does not
support more than one worker.

bandetto committed:
Add a check for CTask::Self() (#863)

CTask::Self() uses dynamic_cast internally and may return NULL. Add a check to
avoid dereferencing a NULL pointer.

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/65175

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1095104

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1095105

BenderArenadata commented:
Failed job Behave tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1095102
